### ISSUE TITLE
Manually added connector cannot get the latest version

### ISSUE THREAD

**Issue by tuliren** [airbytehq/airbyte#6233]
## Current Behavior
If a user adds a connector manually before the connector is released, later on when the connector is release, the user cannot see the latest version for it on the UI.

Slack [thread](https://airbytehq.slack.com/archives/C01A4CAP81L/p1631881619089900?thread_ts=1631838787.083300&cid=C01A4CAP81L):

![image (1)](https://user-images.githubusercontent.com/1933157/133853925-47884c19-a9a4-4680-8331-b93c30f0c0ca.png)

In the screenshot, the `MongoTesting` is the manually added mongodb v2 connector. It has the same docker repository.

## Expected Behavior
Ideally, if a manually added connector is the same as the one in the Airbyte codebase and has a newer version, it can get matched on the UI and user can easily upgrade it.

## Steps to Reproduce
1. In an old Airbyte version, manually add a connector that does not exist in that version, but is released in a future Airbyte version (with the help of a time machine).
2. Upgrade Airbyte to a new version including a new version of the manually added connector.
3. Go to the connector list page.

## Are you willing to submit a PR?
I think the change is probably just to change this code block here:
https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/core/resources/SourceDefinition.ts#L48-L51

```diff
-  latestSource.sourceDefinitionId === source.sourceDefinitionId
+  latestSource.dockerRepository === source.dockerRepository
```

**Comment by tuliren** [comment]
@jrhizor, FYI.

**Comment by jrhizor** [comment]
I don't think that will actually work by itself. The "list latest" endpoint pulls from the Github store, not the Docker registry itself, so it has no way of knowing the latest for a manual connector. Maybe we could add support for that at some point. For docker it'd be pretty easy (even for private repos), but for kubernetes it'd be more difficult.

**Comment by tuliren** [comment]
> it has no way of knowing the latest for a manual connector

As long as the docker repository is the same, we know a manually added connector is one of the official connector. In this way, we can tell a manual connector has new version, right?

**Comment by jrhizor** [comment]
Ah ok, I understand. We would expect this to never match for custom connectors, this is solely for people adding a connector before upgrading Airbyte and ending up with a different definition id.

Moving back to medium.

**Comment by tuliren** [comment]
> this is solely for people adding a connector before upgrading Airbyte and ending up with a different definition id.

Yes.

I suspect that the one line change I proposed in this issue is the only change we need. But I currently don't have bandwidth to verify it.
